### PR TITLE
itemst vmethod update & housecleaning

### PR DESCRIPTION
**Pull request by ab9rf**
identified a significant number of anonymous vmethods in `itemst`

also cleaned up the index numbers in the xml, they were badly out of whack

### CODE REVIEW

File: df.items.xml [hunk 1]
```diff
@@ -623,46 +631,60 @@
...
                 <enum base-type='int16_t' type-name='job_skill' name='best_any'/>
                 <enum base-type='int16_t' type-name='job_skill' name='best_melee'/>
                 <enum base-type='int16_t' type-name='job_skill' name='best_ranged'/>
             </vmethod>
-
-            -- 218
-
             <vmethod ret-type='int32_t' name='calcBaseUniformScore'/>
-            <vmethod><int32_t/><int32_t/><int32_t/></vmethod>
+            <vmethod ret-type='int32_t'> <int32_t name='race_id'/><int32_t name='caste_id'/><int32_t name='maximum'/></vmethod>
```
Comment (myk002) [inline df.items.xml:668]: no name for this one?

File: df.items.xml [hunk 2]
```diff
@@ -623,46 +631,60 @@
                 <pointer name='u' type-name='unit'/>
                 <pointer name='j' type-name='job'/>
             </vmethod>
-            <vmethod><int32_t/><int32_t/><int32_t/><int32_t/><int32_t/><int32_t/><int32_t/><int32_t/><int32_t/><int32_t/><int32_t/><int32_t/></vmethod>
-            <vmethod><int32_t/></vmethod>
+            <vmethod name='addImprovements'>
```
Comment (myk002) [inline df.items.xml:634]: `addImprovement`? I only see one

Comment (ab9rf) [reply df.items.xml:634]: the code may be able to add more than one improvement, and it is parallel with the subsequent method which definitely can add multiple improvements

File: df.items.xml (hunk 1 above)
Comment (ab9rf) [reply df.items.xml:668]: nope. don't know what it does

File: df.items.xml (hunk 2 above)
Comment (ab9rf) [reply df.items.xml:634]: ok, i reviewed the one non-trivial implementation and it only adds one improvement, so we'll drop the 's'

**Review by myk002 — APPROVED** [review]

**Pull request merged**